examples/src/test/scala/docs/KafkaToSlickExample.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+package docs
+
+import scala.concurrent.duration._
+
+import akka.Done
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.adapter._
+import akka.kafka.ConsumerSettings
+import akka.kafka.ProducerSettings
+import akka.kafka.Subscriptions
+import akka.kafka.scaladsl.SendProducer
+import akka.projection.ProjectionId
+import akka.projection.internal.MergeableOffsets
+import akka.projection.kafka.KafkaSourceProvider
+import akka.projection.scaladsl.SourceProvider
+import akka.projection.slick.SlickProjection
+import com.typesafe.config.ConfigFactory
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.apache.kafka.common.serialization.ByteArrayDeserializer
+import org.apache.kafka.common.serialization.ByteArraySerializer
+import org.apache.kafka.common.serialization.StringDeserializer
+import org.apache.kafka.common.serialization.StringSerializer
+import slick.basic.DatabaseConfig
+import slick.dbio.DBIOAction
+import slick.jdbc.H2Profile
+
+object KafkaToSlickExample {
+
+  ActorSystem[Nothing](Guardian(), "KafkaToSlickExample")
+
+  object Guardian {
+    def apply(): Behavior[Nothing] = {
+      Behaviors.setup[Nothing] { context =>
+        implicit val system = context.system
+
+        val config = ConfigFactory.parseString("""
+        akka {
+         loglevel = "DEBUG"
+         projection.slick = {
+  
+           profile = "slick.jdbc.H2Profile$"
+  
+           # TODO: configure connection pool and slick async executor
+           db = {
+            url = "jdbc:h2:mem:test1"
+            driver = org.h2.Driver
+            connectionPool = disabled
+            keepAliveConnection = true
+           }
+         }
+        }
+        """)
+        val dbConfig: DatabaseConfig[H2Profile] = DatabaseConfig.forConfig("akka.projection.slick", config)
+        val projectionId = ProjectionId("ExampleProjection", "ExampleProjection-1")
+        val producerSettings = ProducerSettings(system, new StringSerializer, new ByteArraySerializer)
+        val sendProducer = SendProducer(producerSettings)(system.toClassic)
+        val consumerSettings = ConsumerSettings(system, new StringDeserializer, new ByteArrayDeserializer)
+        val kafkaSourceProvider: SourceProvider[MergeableOffsets.Offset, ConsumerRecord[String, Array[Byte]]] =
+          KafkaSourceProvider(consumerSettings, Subscriptions.topics("user-events"))
+
+        val projection =
+          SlickProjection.atLeastOnce(
```

Comment:
it only adds value for the exactlyOnce case, otherwise it's better to commit offsets back to Kafka

Reply:
I think users would be interested in storing Kafka offsets in more than just the slick transactional use case.  Users might want to externally control offsets regardless of message delivery semantics.  For example, they may they have an internal system tracking or even updating the offset store, they may have relaxed retention config for the internal consumer offsets Kafka topic, or maybe they just treat all Kafka data as ephemeral and don't want to use it for long term storage of offset information.

I've updated the example to use `SlickProjection.exactlyOnce`, but I don't believe there's any reason we should recommend that the user not use `atLeastOnce`.  I think we should support `MergeableOffset` in Cassandra and other implementations in the future too.

Reply:
good point 👍

Reply:
Didn't see this before, please create a ticket about this.

Reply:
https://github.com/akka/akka-projection/issues/97